DopplerCurrencyJob/DopplerCurrencyJob.cs

```diff
@@ -25,27 +29,85 @@ public class DopplerCurrencyJob
-        public IList<CurrencyResponse> Run() => RunAsync().GetAwaiter().GetResult();
+        public DopplerCurrencyJobResponse Run() => RunAsync().GetAwaiter().GetResult();
 
-        private async Task<IList<CurrencyResponse>> RunAsync()
+        private async Task<DopplerCurrencyJobResponse> RunAsync()
         {
-            _logger.LogInformation("Getting currency per each code enabled.");
+            var taskFailed = false;
+            var resultBuffer = new StringBuilder();
+            LogAndAppendToStringBuilder(resultBuffer, "Getting currency per each code enabled.");
+
             var currencyDto = await _dopplerCurrencyService.GetCurrencyByCode();
 
             if (!currencyDto.Any())
             {
-                _logger.LogInformation("Non-existent currencies for this date, please check for errors.");
-                return new List<CurrencyResponse>();
+                LogAndAppendToStringBuilder(resultBuffer, "Non-existent currencies for this date, please check for errors.");
+                return new DopplerCurrencyJobResponse()
+                {
+                    CurrencyList = new List<CurrencyResponse>(),
+                    Message = resultBuffer.ToString()
+                };
+            }
+
+            LogAndAppendToStringBuilder(resultBuffer, "Currency retrieved by code:");
+            AppendSerializedToStringBuilder(resultBuffer, currencyDto);
+
+            try
+            {
+                LogAndAppendToStringBuilder(resultBuffer, "Sending currency data to Doppler SAP system.");
+                var result = await _dopplerSapService.SendCurrency(currencyDto);
+
+                if (!result.IsSuccessStatusCode)
+                {
+                    LogAndAppendToStringBuilder(resultBuffer, $"{result.ReasonPhrase}. Error sending currency to Doppler SAP Api.");
+                    taskFailed = true;
+                }
+
+                LogAndAppendToStringBuilder(resultBuffer, "Sent currency data to Doppler SAP system.");
+            }
+            catch (Exception e)
+            {
+                LogAndAppendToStringBuilder(resultBuffer, $"An exception occurred when sending currency to Doppler SAP Api. Exception: {e.Message}");
+                taskFailed = true;
             }
 
-            _logger.LogInformation("Sending currency data to Doppler SAP system.");
-            await _dopplerSapService.SendCurrency(currencyDto);
+            try
+            {
+                LogAndAppendToStringBuilder(resultBuffer, "Insert currency data into Doppler Database.");
+
+                await _dopplerCurrencyService.InsertCurrencyIntoDataBase(currencyDto);
+
+                LogAndAppendToStringBuilder(resultBuffer, "Inserted currency data into Doppler Database.");
+            }
+            catch (Exception e)
+            {
+                LogAndAppendToStringBuilder(resultBuffer, $"An exception occurred when Insert currency data into Doppler Database. Exception: {e.Message}");
```

Comment:
Creo que esto hay que loguearlo como error

```suggestion
                LogAndAppendErrorToStringBuilder(resultBuffer, "An exception occurred when Insert currency data into Doppler Database.", e);
```

Reply:
I change the level of some logs is changed and the helper method is improved to handle different levels of logs

Reply:
Sigue acá: https://github.com/MakingSense/doppler-jobs/pull/84/files#discussion_r529557758

Marco como resuelto.